api/src/services/users/create_api_key.py

```diff
@@ -59,6 +84,45 @@ def create_api_key(db_session: db.Session, user_id: UUID, json_data: dict) -> Us
     return api_key
 
 
+def _import_api_key_to_aws_gateway(api_key: UserApiKey) -> None:
+    """Import an API key to AWS API Gateway and associate it with a usage plan"""
+    try:
+        config = ApiGatewayConfig()
+
+        gateway_response = import_api_key(
+            api_key=api_key.key_id,
+            name=api_key.key_name,
+            description=f"API key for user {api_key.user_id}",
+            enabled=api_key.is_active,
+            usage_plan_id=config.default_usage_plan_id,
+        )
+
+        if config.default_usage_plan_id:
+            logger.info(
+                "Successfully imported API key to AWS API Gateway and associated with usage plan",
+                extra={
+                    "api_key_id": api_key.api_key_id,
+                    "gateway_key_id": gateway_response.id,
+                    "usage_plan_id": config.default_usage_plan_id,
+                },
+            )
+        else:
+            logger.info(
+                "Successfully imported API key to AWS API Gateway (no usage plan configured)",
+                extra={
+                    "api_key_id": api_key.api_key_id,
+                    "gateway_key_id": gateway_response.id,
+                },
+            )
```

Comment:
Is there a scenario where we wouldn't have a default usage plan ID defined? Should the value be nullable?

It's fine if we have to define a value locally even though it doesn't talk to AWS. Just a string in local.env

Reply:
Why do you recommend commenting it out? Having the value just set in local.env and removing these extra code-paths feels simpler / better setup for developers.